Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,591 @@
+                for numor in numors:
+                    if average_detectors:
+                        try:
+                            Plus(LHSWorkspace=name, RHSWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                        except ValueError:
+                            CloneWorkspace(InputWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                    else:
+                        list_pol.append('{0}_{1}'.format(numor, direction))
+                if average_detectors:
+                    CreateSingleValuedWorkspace(DataValue=len(numors), OutputWorkspace='norm')
+                    Divide(LHSWorkspace=name, RHSWorkspace='norm', OutputWorkspace=name)
+                    DeleteWorkspace(Workspace='norm')
+                else:
+                    SumOverlappingTubes(','.join(list_pol), OutputWorkspace=name,
+                                        OutputType='1D', ScatteringAngleBinning=self.getProperty('ScatteringAngleBinSize').value,
+                                        Normalise=True, HeightAxis='-0.1,0.1')
+                names_list.append(name)
+            DeleteWorkspaces(WorkspaceList=ws)
+            GroupWorkspaces(InputWorkspaces=names_list, OutputWorkspace=ws)
+        return ws
+
+    def _normalise(self, ws):
+        """Normalises the provided WorkspaceGroup to the monitor 1."""
+        for entry_no, entry in enumerate(mtd[ws]):
+            mon = ws + '_mon'
+            ExtractMonitors(InputWorkspace=entry, DetectorWorkspace=entry,
+                            MonitorWorkspace=mon)
+            if 0 in mtd[mon].readY(0):
+                raise RuntimeError('Cannot normalise to monitor; monitor has 0 counts.')
+            else:
+                CreateSingleValuedWorkspace(DataValue=mtd[mon].readY(0)[0], ErrorValue=mtd[mon].readE(0)[0],
+                                            OutputWorkspace=mon)
+                Divide(LHSWorkspace=entry, RHSWorkspace=mon, OutputWorkspace=entry)
+                DeleteWorkspace(Workspace=mon)
+        return ws
+
+    def _calculate_transmission(self, ws, beam_ws):
+        """Calculates transmission based on the measurement of the current sample and empty beam."""
+        # extract Monitor2 values
+        if 0 in mtd[ws][0].readY(0):
+            raise RuntimeError('Cannot calculate transmission; monitor has 0 counts.')
+        if 0 in mtd[beam_ws].readY(0):
+            raise RuntimeError('Cannot calculate transmission; beam monitor has 0 counts.')
+        Divide(LHSWorkspace=ws, RHSWorkspace=beam_ws, OutputWorkspace=ws)
+        return ws
+
+    def _subtract_background(self, ws, container_ws, transmission_ws):
+        """Subtracts empty container and absorber scaled by transmission."""
+        absorber_ws = self.getPropertyValue('AbsorberInputWorkspace')
+        if absorber_ws == "":
+            raise RuntimeError("Absorber input workspace needs to be provided for monochromatic background subtraction.")
```

Comment:
Is this really mandatory, I don't understand why.

Reply:
It can be a return statement instead to cut off running background subtraction without the necessary inputs. The absorber_ws is not a parameter of this method due to expected workflow when TOF is implemented.